examples/modular-drivers/src/lib.rs

```diff
@@ -1,10 +1,16 @@
 use micro_rdk::common::registry::{ComponentRegistry, RegistryError};
 
+#[cfg(feature = "esp32")]
 pub mod free_heap_sensor;
+pub mod moisture_sensor;
+pub mod water_pump;
+#[cfg(feature = "esp32")]
 pub mod wifi_rssi_sensor;
 
 pub fn register_models(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
-    wifi_rssi_sensor::register_models(registry)?;
     free_heap_sensor::register_models(registry)?;
+    moisture_sensor::register_models(registry)?;
+    water_pump::register_models(registry)?;
+    wifi_rssi_sensor::register_models(registry)?;
```

Comment:
the wifi_rssi_sensor and free_heap_sensor should also be feature gated here for esp32. Otherwise, running this on native would not compile.